### DISCUSSION TITLE
市のオープンデータにどうやって緯度経度を付与するか

### DISCUSSION THREAD

**Discussion by aca-ta** [Code-for-Funabashi/map#6] [General]
市で公開しているオープンデータには緯度経度が無いため、そのままでは地図上にマッピングできない。
どうやって緯度経度情報を取得するかを考える。

データの一例
https://www.city.funabashi.lg.jp/opendata/002/p059791.html

**Comment by j-takurou** [comment]
叩き台案として、
- 施設情報の住所があるので~~逆ジオコーディング~~ジオコーディングしていく。ジオコーディングする際の情報として
     - https://nlftp.mlit.go.jp/cgi-bin/isj/dls/_download_files.cgi
         - 下記のような対応表がダウンロードできました 
         - あ、でもこれ住所情報の粒度粗すぎますね...
     - google reverse-geocode api（みたいなやつあったような...?）


都道府県コード | 都道府県名 | 市区町村コード | 市区町村名 | 大字町丁目コード | 大字町丁目名 | 緯度 | 経度
-- | -- | -- | -- | -- | -- | -- | --
12 | 千葉県 | 12204 | 船橋市 | 1.2204E+11 | 旭町 | 35.731294 | 139.984841
12 | 千葉県 | 12204 | 船橋市 | 1.2204E+11 | 旭町一丁目 | 35.727994 | 139.986361
12 | 千葉県 | 12204 | 船橋市 | 1.2204E+11 | 旭町二丁目 | 35.735 | 139.987414

**Reply by aca-ta** [reply]
> 施設情報の住所

ポイントデータレベルだと国交省のデータでは賄えないので、Googleとかのサービスを使うしかないかなと思っています。
無料かつ規約的に自由に使える物を調査をしないとなーという課題感を持っています！

**Reply by j-takurou** [reply]
そうですね、ここら辺明日調査してみます！

survey発見.!
https://deerfoot.exblog.jp/21495211/

認識合わせたいのですが、”閲覧のタイミングでgeocode apiを叩く”のではなく、今リストとして上がっている施設のlatlonがわかれば良いと言う認識ですよね？
であれば、現状の船橋の施設数的にgoogle apiの月無料枠に収まる気がしました。（40000reqまで無料とのこと
<img width="835" alt="スクリーンショット 2021-03-11 2 39 27" src="https://user-images.githubusercontent.com/29911671/110672472-02544a00-8213-11eb-91d3-afe0090e3070.png">
）

**Reply by aca-ta** [reply]
> ”閲覧のタイミングでgeocode apiを叩く”のではなく、今リストとして上がっている施設のlatlonがわかれば良い

そうです！別途バッチ等で事前にデータ作成する想定でした。
[Google Geocoding API](https://developers.google.com/maps/documentation/geocoding/overview)で良さそうですね！
